ui/src/views/Clients/ViewClient/index.jsx

```diff
@@ -165,35 +163,46 @@ export default class ViewClient extends Component {
     this.setState({ error: null, loading: true });
 
     try {
-      await this.props.client.mutate({
+      const result = await this.props.client.mutate({
         mutation: isNewClient ? createClientQuery : updateClientQuery,
         variables: {
           clientId,
           client,
         },
       });
 
-      this.setState({ error: null, loading: false });
+      this.setState({
+        error: null,
+        loading: false,
+      });
 
       if (isNewClient) {
-        this.props.history.push(
-          `/auth/clients/${encodeURIComponent(clientId)}`
-        );
+        this.props.history.push({
+          pathname: `/auth/clients/${encodeURIComponent(clientId)}`,
+          state: { accessToken: result.data.createClient.accessToken },
+        });
       }
     } catch (error) {
       this.setState({ error, loading: false });
     }
   };
 
   render() {
-    const { error, loading, accessToken, accessTokenPanelOpen } = this.state;
-    const { isNewClient, data, classes } = this.props;
+    const { error, loading, accessToken } = this.state;
+    const { isNewClient, data, classes, location } = this.props;
+
+    if (location.state && location.state.accessToken) {
+      const state = { ...location.state };
+
+      delete state.accessToken;
```

Comment:
Isn't `location` supposed to be immutable?

Reply:
It is, hence the `{ ...location.state }`.